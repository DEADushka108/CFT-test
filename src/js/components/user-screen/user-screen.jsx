import React, {Fragment, useEffect} from 'react';
import PropTypes from 'prop-types';
import Header from '../header/header';
import Footer from '../footer/footer';
import {getActiveUser, getUserPosts} from '../../store/users/selectors';
import {Operation as UsersOperation} from '../../store/users/users';
import {connect} from 'react-redux';
import {userDetails} from '../../types/user';
import {postDetails} from '../../types/post';
import UserCardSmall from '../user-card-small/user-card-small';
import {Link} from 'react-router-dom';
import {AppRoute} from '../../utils/const';

const UserScreen = (props) => {
  const {userInfo, userPosts, match, onLoadPosts, onLoadUser} = props;
  const routeId = Number(match.params.id);

  useEffect(() => {
    onLoadUser(routeId);
    onLoadPosts(routeId);
  }, [routeId]);


  return <Fragment>
    <Header/>
    <main className="page-main">
      <h1 className="visually-hidden">User</h1>
      <section className="user-block">
        <h2 className="user-block__title">User information:</h2>
        <UserCardSmall userInfo={userInfo}/>
      </section>
      <section className="user-posts">
        <h2 className="user-posts__title">User posts:</h2>
        <ul className="user-posts__list">
          {userPosts.map((post, index) => {
            const {id, title} = post;
            return <li key={id} className="user-posts__item">
              <Link to={`${AppRoute.POSTS}/${id}`} className="user-posts__link">{index + 1}. {title}</Link>
            </li>;
          })}
        </ul>
      </section>
    </main>
    <Footer/>

  </Fragment>;
};

UserScreen.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }),
  }).isRequired,
  userInfo: userDetails,
  userPosts: PropTypes.arrayOf(postDetails),
  onLoadPosts: PropTypes.func.isRequired,
  onLoadUser: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  userInfo: getActiveUser(state),
  userPosts: getUserPosts(state),
});

const mapDispatchToProps = (dispatch) => ({
  onLoadUser(id) {
    dispatch(UsersOperation.loadUser(id));
  },
  onLoadPosts(id) {
    dispatch(UsersOperation.loadUserPosts(id));
  },
});

export {UserScreen};
export default connect(mapStateToProps, mapDispatchToProps)(UserScreen);
