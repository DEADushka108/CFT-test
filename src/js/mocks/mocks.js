export * from './const/const';
export * from './posts/posts';
export * from './store/store';
export * from './users/users';
export * from './comments/comments';
