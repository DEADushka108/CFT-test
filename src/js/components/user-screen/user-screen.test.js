import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {UserScreen} from './user-screen';
import {store, match, user, noop, postsList} from '../../mocks/mocks.js';

it(`Should render UserScreen component correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <UserScreen
            userInfo={user}
            userPosts={postsList}
            match={match}
            onLoadPosts={noop}
            onLoadUser={noop}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    });

  expect(tree).toMatchSnapshot();
});
