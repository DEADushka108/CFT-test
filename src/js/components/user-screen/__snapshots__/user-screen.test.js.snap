// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render UserScreen component correctly 1`] = `
Array [
  <header
    className="page-header"
  >
    <div
      className="page-header__wrapper"
    >
      <a
        className="page-header__logo"
        href="/"
        onClick={[Function]}
      >
        Chirrup
      </a>
      <nav
        className="main-nav"
      >
        <ul
          className="main-nav__list"
        >
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link main-nav__link--active"
              href="/"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#home-page"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Main
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/users"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#social-group"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Users
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/contacts"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#message"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Contact us
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>,
  <main
    className="page-main"
  >
    <h1
      className="visually-hidden"
    >
      User
    </h1>
    <section
      className="user-block"
    >
      <h2
        className="user-block__title"
      >
        User information:
      </h2>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Bret"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Leanne Graham
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/1"
                onClick={[Function]}
              >
                Bret
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Sincere@april.biz"
              >
                Sincere@april.biz
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:1-770-736-8031 x56442"
              >
                1-770-736-8031 x56442
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="hildegard.org"
              >
                hildegard.org
              </a>
            </li>
          </ul>
        </div>
      </article>
    </section>
    <section
      className="user-posts"
    >
      <h2
        className="user-posts__title"
      >
        User posts:
      </h2>
      <ul
        className="user-posts__list"
      >
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/1"
            onClick={[Function]}
          >
            1
            . 
            sunt aut facere repellat provident occaecati excepturi optio reprehenderit
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/2"
            onClick={[Function]}
          >
            2
            . 
            qui est esse
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/3"
            onClick={[Function]}
          >
            3
            . 
            ea molestias quasi exercitationem repellat qui ipsa sit aut
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/4"
            onClick={[Function]}
          >
            4
            . 
            eum et est occaecati
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/5"
            onClick={[Function]}
          >
            5
            . 
            nesciunt quas odio
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/6"
            onClick={[Function]}
          >
            6
            . 
            dolorem eum magni eos aperiam quia
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/7"
            onClick={[Function]}
          >
            7
            . 
            magnam facilis autem
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/8"
            onClick={[Function]}
          >
            8
            . 
            dolorem dolore est ipsam
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/9"
            onClick={[Function]}
          >
            9
            . 
            nesciunt iure omnis dolorem tempora et accusantium
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/10"
            onClick={[Function]}
          >
            10
            . 
            optio molestias id quia eum
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/11"
            onClick={[Function]}
          >
            11
            . 
            et ea vero quia laudantium autem
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/12"
            onClick={[Function]}
          >
            12
            . 
            in quibusdam tempore odit est dolorem
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/13"
            onClick={[Function]}
          >
            13
            . 
            dolorum ut in voluptas mollitia et saepe quo animi
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/14"
            onClick={[Function]}
          >
            14
            . 
            voluptatem eligendi optio
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/15"
            onClick={[Function]}
          >
            15
            . 
            eveniet quod temporibus
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/16"
            onClick={[Function]}
          >
            16
            . 
            sint suscipit perspiciatis velit dolorum rerum ipsa laboriosam odio
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/17"
            onClick={[Function]}
          >
            17
            . 
            fugit voluptas sed molestias voluptatem provident
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/18"
            onClick={[Function]}
          >
            18
            . 
            voluptate et itaque vero tempora molestiae
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/19"
            onClick={[Function]}
          >
            19
            . 
            adipisci placeat illum aut reiciendis qui
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/20"
            onClick={[Function]}
          >
            20
            . 
            doloribus ad provident suscipit at
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/21"
            onClick={[Function]}
          >
            21
            . 
            asperiores ea ipsam voluptatibus modi minima quia sint
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/22"
            onClick={[Function]}
          >
            22
            . 
            dolor sint quo a velit explicabo quia nam
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/23"
            onClick={[Function]}
          >
            23
            . 
            maxime id vitae nihil numquam
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/24"
            onClick={[Function]}
          >
            24
            . 
            autem hic labore sunt dolores incidunt
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/25"
            onClick={[Function]}
          >
            25
            . 
            rem alias distinctio quo quis
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/26"
            onClick={[Function]}
          >
            26
            . 
            est et quae odit qui non
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/27"
            onClick={[Function]}
          >
            27
            . 
            quasi id et eos tenetur aut quo autem
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/28"
            onClick={[Function]}
          >
            28
            . 
            delectus ullam et corporis nulla voluptas sequi
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/29"
            onClick={[Function]}
          >
            29
            . 
            iusto eius quod necessitatibus culpa ea
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/30"
            onClick={[Function]}
          >
            30
            . 
            a quo magni similique perferendis
          </a>
        </li>
        <li
          className="user-posts__item"
        >
          <a
            className="user-posts__link"
            href="/posts/31"
            onClick={[Function]}
          >
            31
            . 
            ullam ut quidem id aut vel consequuntur
          </a>
        </li>
      </ul>
    </section>
  </main>,
  <footer
    className="page-footer"
  >
    <div
      className="page-footer__wrapper"
    >
      <a
        className="page-footer__logo"
        href="/"
        onClick={[Function]}
      >
        Chirrup
      </a>
      <div
        className="page-footer__copyright"
      >
        <p
          className="page-footer__text"
        >
          © 2020 Chirrup Ltd.
        </p>
      </div>
    </div>
  </footer>,
]
`;
